### PR TITLE
pip-tools

### PR DESCRIPTION
**Pull request by Br4nd0R**
# Changes
* We now have two new `Makefile` targets: `update-deps` and `dev-install`. The prior updates dependencies via `pip-compile` and the latter installs the requirements via `pip-sync`.

We have pinned several dependency versions to fix various tests that fail if they are upgraded to their latest versions.

### CODE REVIEW

File: requirements/dev.in [hunk 1]
```diff
@@ -0,0 +1,21 @@
+-c prod.txt
+-e .
```
Comment (davisagli) [inline requirements/dev.in:2]:
```suggestion
```

This directory does not contain a Python package that we want to install in editable form.

**Comment by davisagli** [comment]
> Running pip-sync requirements/*.txt works but pip-sync requirements_dev.txt throws an error that I'm unsure of how to resolve.

What's the error?

> I also didn't see a line for cumulusci== in dev.txt. I'm not sure if this is expected or not, but we did see it referenced with a @git tag when we pip-compiled without hashes before, so thought it would be worth mentioning.

That's expected. We're producing a list of packages that need to be installed to do development on CumulusCI, but generally the user performing the installation is picking what version of cumulusci to install (either by pip installing a particular version, or pip installing requirements_dev.txt which uses `-e .` to refer to the currently checked out git commit.

> Also, should I update the .github/workflows/* to use pip-sync or do we want to ensure that pip install works too?

Let's try it.

File: utility/build-homebrew.sh [hunk 2]
```diff
@@ -48,7 +48,7 @@ $(cat "$REQS_FILE")
 REQS
 
     File.write("requirements.txt", reqs)
-    system "python3", "-m", "pip", "install", "-r", "requirements.txt", "--require-hashes", "--no-deps", "--ignore-installed", "--prefix", libexec
+    system "python3", "-m", "pip", "install", "-r", "requirements.txt", "--no-deps", "--ignore-installed", "--prefix", libexec
```
Comment (davisagli) [inline utility/build-homebrew.sh:51]: @Br4nd0R I don't think we want to remove it here; the homebrew installer is for Macs so we should be able to rely on it including the correct dependencies for the platform as long as we're building the homebrew formula from a Mac.

File: Makefile [hunk 3]
```diff
@@ -94,3 +94,11 @@ install: clean ## install the package to the active Python's site-packages
 tag: clean
 	git tag -a -m 'version $$(python setup.py --version)' v$$(python setup.py --version)
 	git push --follow-tags
+
+update-deps:
+	pip-compile --upgrade --allow-unsafe requirements/prod.in
+	pip-compile --upgrade --upgrade-package 'typeguard==2.10.0' --allow-unsafe requirements/dev.in
```
Comment (davisagli) [inline Makefile:100]: typeguard==2.10.0 should be in requirements/dev.in, not here

Comment (Br4nd0R) [reply Makefile:100]: looks like we need to hold back `robot` too

Comment (davisagli) [reply Makefile:100]: Ah yes -- both robotframework and robotframework-seleniumlibrary

**Comment by Br4nd0R** [comment]
@davisagli - This should be (hopefully) good-to-go now.

File: requirements/prod.txt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    #   simple-salesforce
+robotframework-lint==1.1
+    # via -r requirements/prod.in
+robotframework-pabot==1.11
+    # via -r requirements/prod.in
+robotframework-pythonlibcore==2.2.1
+    # via robotframework-seleniumlibrary
+robotframework-requests==0.8.1
+    # via -r requirements/prod.in
+robotframework-seleniumlibrary==5.1.3
```
Comment (davisagli) [inline requirements/prod.txt:91]: We shouldn't be updating robotframework-seleniumlibrary to 5.x until Bryan has a look.

File: utility/build-homebrew.sh (hunk 2 above)
Comment (davisagli) [reply utility/build-homebrew.sh:51]: @Br4nd0R Why is this marked resolved?

**Review by davisagli — CHANGES_REQUESTED** [review]

File: requirements/prod.in [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+github3.py
+jinja2
+keyring
+lxml
+pydantic
+PyJWT
+pytz
+requests
+requests-futures
+robotframework==3.2.2
```
Comment (davisagli) [inline requirements/prod.in:15]:
```suggestion
robotframework<4
```

That way we still get minor & patch updates when we recompile, but not the major update until we've reviewed it.

File: utility/build-homebrew.sh (hunk 2 above)
Comment (Br4nd0R) [reply utility/build-homebrew.sh]: Sorry for resolving that thread. I was making these changes late last night, not sure what happened there. I've re-included the `--require-hashes` option in the `build-homebrew` script now.

File: requirements/prod.in [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+robotframework-pabot
+robotframework-requests
+robotframework-seleniumlibrary<5
+rst2ansi
+salesforce-bulk
+sarge
+selenium
+simple-salesforce
+snowfakery
+SQLAlchemy==1.3.23
```
Comment (davisagli) [inline requirements/prod.in:26]:
```suggestion
SQLAlchemy<1.4
```

**Review by davisagli — APPROVED** [review]

**Pull request merged**